src/services/wallets/metamask/metamask.ts

```diff
@@ -0,0 +1,25 @@
+import Web3 from "web3";
+import { HexString } from "../../../utilities/types";
+import ethereum, { EthereumProvider } from "./ethereum";
+
+export const mmweb3 = new Web3(ethereum as EthereumProvider);
```

Comment:
Hmm... the as here is covering up the case where ethereum is undefined or null. We should do that check instead of hiding it. (This might require throwing an error in the case of `getWeb3`)

Reply:
Yeah this will error out currently if someone attempts to log in with Metamask but Metamask isn't installed. I can have it throw an error that Metamask isn't installed. It'll be caught be the login and display as an alert by the login button.

Reply:
Hmm I'm no 100% happy with my solution but it can no longer go through without being checked and throwing an error if it's null or undefined.